### DISCUSSION TITLE
add a frature "sqlite-bundled" for diesel

### DISCUSSION THREAD

**Discussion by yykkxk** [diesel-rs/diesel#3255] [Ideas]
Sqlite3 doesn't offer precompiled **sqlite.lib** for Windows 10, so diesel ( with the feature "sqlite") always build failed.
Now diesel has a feature: **sqlite = ["libsqlite3-sys", "diesel_derives/sqlite"]**, and I try to change the feature as this: **sqlite = ["libsqlite3-sys/bundled", "diesel_derives/sqlite"]**, then libsqlite3-sys will compile SQLite from source and link against that,  and the problem is resolved.
Can we add a new feature for diesel like this:  **sqlite-bundled = ["libsqlite3-sys/bundled", "diesel_derives/sqlite"]**, then  **sqlite.lib** has not to be installed  on Windows 10(or Win11).

**Comment by weiznich** [comment]
Can you elaborate on why this needs an extra feature in diesel? You can just add the corresponding dependency by yourself and enable the feature flag there. Cargo feature flags are additive, so if your crate enables the `bundled` feature for `libsqlite3-sys` that will also apply to diesel as well. I personally see no meaning in adding a essentially duplicated feature for this to diesel.

**Reply by yykkxk** [reply]
Install sqlite.lib on Linux is easy, but many people have problems using diesel on windows. 
SQLite3 officially only offers sqlite.dll ( instead of sqlite.lib) for Windows, so diesel (feature "sqlite") is difficult to use on Windows.  Someone must compile the source code of sqlite to get sqlite.lib. 
So why doesn't diesel enable the feature "bundled" for sqlite(provided by libsqlite-sys)?
 I find diesel_cli has a feature  "sqlite-bundled" to solve this problem.

**Reply by weiznich** [reply]
Please read my answer again. It's not about linux or any other operating system at all. There is no need for having such feature on diesel itself, as **you** as a user can control that behaviour already on yourself. That's not the case for diesel-cli as you don't control the `Cargo.toml` file there.

Defaulting to the bundled feature is also no option as there are valid use cases for not enabling this feature too.

**Reply by schlamar** [reply]
@weiznich Maybe you should consider updating the documentation as right now this somehow implies that bundling SQLite is not possible:

> sqlite: This feature enables the diesel sqlite backend. Enabling this feature requires a compatible copy of libsqlite3 for your target architecture.

**Reply by weiznich** [reply]
@schlamar I'm happy to merge a PR for such a change.
